From 86ae4336772fd71fb2e679a239d61b7868df5bd4 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 23 Dec 2024 22:46:33 +0100
Subject: [PATCH 1/3] setup-wayland-base: explicit start cgroups and dbus

Seems like openrc does not pull in dependencies when starting services
from other rc service (eg tiny-cloud). Work around it by explicitly
start cgroup and dbus first.

Also, do not autostart elogind or polkit. Dbus will autostart those when
needed.
---
 setup-wayland-base.in         | 2 +-
 tests/setup_wayland_base_test | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup-wayland-base.in b/setup-wayland-base.in
index 267c835..8f0138c 100644
--- a/setup-wayland-base.in
+++ b/setup-wayland-base.in
@@ -30,7 +30,7 @@ apk add elogind polkit-elogind eudev "$@"
 
 setup-devd udev
 
-for service in elogind polkit; do
+for service in cgroups dbus; do
 	rc-service "$service" start
 	rc-update add "$service"
 done
diff --git a/tests/setup_wayland_base_test b/tests/setup_wayland_base_test
index f7e9e68..b7fd2e7 100755
--- a/tests/setup_wayland_base_test
+++ b/tests/setup_wayland_base_test
@@ -21,7 +21,7 @@ setup_wayland_base_body() {
 	atf_check -s exit:0 \
 		-o match:"apk add.*elogind polkit-elogind.*foobar" \
 		-o match:"Starting udev" \
-		-o match:"Starting polkit" \
+		-o match:"Starting dbus" \
 		-e empty \
 		setup-wayland-base foobar
 
-- 
2.47.1

